src/test/java/org/testcontainers/jooq/codegen/PluginTest.java

```diff
@@ -1,77 +1,84 @@
 package org.testcontainers.jooq.codegen;
 
-import static assertions.SchemaAssert.assertThatDefaultSchema;
-import static assertions.SchemaAssert.assertThatSchema;
-import static common.Common.GENERATED_ROOT_DIR;
+import static assertions.MavenProjectAssert.assertThatProject;
+import static org.codehaus.plexus.testing.PlexusExtension.getTestFile;
 
-import java.io.File;
-import org.apache.commons.io.FileUtils;
-import org.apache.maven.plugin.testing.AbstractMojoTestCase;
+import org.apache.maven.plugin.testing.MojoRule;
+import org.apache.maven.project.MavenProject;
+import org.junit.Rule;
+import org.junit.Test;
 
 /**
- * Integration test plugin under real pom files
+ * Integration test plugin under real pom files <br/>
+ * Sources by default are generated under the same test pom directory <br/>
+ * You can check them after test runs, nothing gets into final jar
  */
-public class PluginTest extends AbstractMojoTestCase {
-
-    /**
-     * Cleanup generated sources
-     */
-    @Override
-    public void tearDown() throws Exception {
-        super.tearDown();
-        FileUtils.deleteDirectory(GENERATED_ROOT_DIR);
-    }
+public class PluginTest {
+
+    @Rule
```

Comment:
Is it not possible to use JUnit 5 here? Once we introduce the dependency on JUnit 4 it will be hard to remove later.

Reply:
Actually `AbstractMojoTestCase` extends `TestCase` from **Junit3** .
Usage of `@Rule` make possible to use **Junit4** out of the box. I agree that v5 better than v4, but  generally there is no difference ,because we have `junit-vintage-engine` which gets us backward compatibility.
I'd love to pick Junit5 ,but `maven-plugin-testing-harness` does not support it :(